tools/start-local-vm

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env bash
+
+shopt -s nullglob
+
+arch=${BUILDSYS_ARCH}
+variant=${BUILDSYS_VARIANT}
+vm_mem=4G
+vm_cpus=4
+force_extract=
+declare -A extra_files=()
+
+current_images=()
+boot_image=
+data_image=
+
+readonly repo_root=$(git rev-parse --show-toplevel)
+
+bail() {
+    >&2 echo "$@"
+    exit 1
+}
+
+show_usage() {
+    local error=$1
+    local newline=$'\n'
+    local usage="\
+usage: ${0##*/} [--arch BUILDSYS_ARCH] [--variant BUILDSYS_VARIANT]
+                      [--vm-memory VM_MEMORY] [--vm-cpus VM_CPUS]
+                      [--inject-file LOCAL_PATH[:IMAGE_PATH]]...
+
+Launch a local virtual machine from a Bottlerocket image.
+
+Options:
+
+    --arch              architecture of the Bottlerocket image (must match the
+                        host architecture ($(uname -m)); may be omitted if the
+                        BUILDSYS_ARCH environment variable is set)
+    --variant           Bottlerocket variant to run (may be omitted if the
+                        BUILDSYS_VARIANT environment variable is set)
+    --vm-memory         amount of memory to assign to the VM; VM_MEMORY must be
+                        a valid QEMU memory specifier (default is ${vm_mem})
+    --vm-cpus           number of CPUs to spawn for VM (default is ${vm_cpus})
+    --force-extract     force recreation of the extracted Bottlerocket image,
+                        e.g. to incorporate new --inject-file options
+    --inject-file       adds a local file to the private partition of the
+                        Bottlerocket image before launching the virtual machine
+                        (may be given multiple times)
+    --help              shows this usage text
+
+The virtual machine's port 22 (SSH) will be exposed via the local port 2222,
```

Comment:
Should the SSH host port mapping be an argument?

Reply:
In case the host's port 2222 is already taken, this would be helpful, yes. It won't do anything to be able to run multiple VMs off the script since there's other work needed for that (e.g. switching to a separate layer for disk writes so VMs don't share their images).

I think a separate `--host-port-forwards` option defaulting to `tcp::2222:22` is most generic and avoids tying this option to SSH too much while still keeping the base case simple.

Reply:
Converted to `--host-port-forwards` with the default as described.